packages/code-du-travail-modeles/src/modeles/conventions/292_plasturgie/salary.ts

```diff
@@ -0,0 +1,59 @@
+import type {
+  IReferenceSalary,
+  ReferenceSalaryProps,
+  SupportedCcIndemniteLicenciement,
+} from "../../common";
+import { nonNullable, rankByMonthArrayDescFrench, sum } from "../../common";
+
+export class ReferenceSalary292
+  implements IReferenceSalary<SupportedCcIndemniteLicenciement.IDCC0292>
+{
+  /**
+   * "- soit S/12
+   * S : total des salaires perçus lors des 12 derniers mois précédant le jour de l'envoi de la lettre de licenciement (brut)
+   *
+   * - soit (S + ((P/12)*3))/3
+   * S : total des salaires perçus lors des 3 derniers mois précédant le jour de l'envoi de la lettre de licenciement (brut)
+   * P : prime(s) ou gratification(s) de caractère annuel ou exceptionnel versée(s) pendant cette période (prise en compte prorata temporis)
+   *
+   * - soit S + (P/12)
+   * S : salaire brut perçu lors du dernier mois précédant le jour de l'envoi de la lettre de licenciement (brut)
+   * P : prime(s) ou gratification(s) de caractère annuel ou exceptionnel versée(s) pendant cette période (prise en compte prorata temporis)"
+   */
+  computeReferenceSalary({
+    salaires = [],
+  }: ReferenceSalaryProps<SupportedCcIndemniteLicenciement.IDCC0292>): number {
+    const rankedSalaires = rankByMonthArrayDescFrench(salaires);
+    const salaryValues = rankedSalaires.map((a) => a.value).filter(nonNullable);
+    const salaireMoyen12DerniersMois = sum(salaryValues) / salaryValues.length;
+
+    const last3salaries = rankedSalaires
+      .filter((s) => nonNullable(s.value))
+      .slice(0, 3);
+
+    const totalLast3SalariesWithoutPrime = sum(
+      last3salaries.map(
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        (s) => s.value! - (s.prime ?? 0)
+      )
+    );
```

Comment:
```suggestion
    const last3salaries = rankedSalaires
      .filter((s) => nonNullable(s.value))
      .slice(0, 3) as (SalaryPeriods & { value: number })[];

    const totalLast3SalariesWithoutPrime = sum(
      last3salaries.map((s) => s.value - (s.prime ?? 0))
    );
```

J'aurais plus caster le type au moment du filter afin de ne plus avoir à gérer le possible null plus bas. Je ne sais pas avec typescript on peut lui faire inférer le fait que la valeur n'est plus null via le filter nonNullable ce qui serait encore mieux.

Reply:
top ! merci ! du coup je l'ai fixé ici aussi : https://github.com/SocialGouv/code-du-travail-numerique/pull/5214/files#diff-4f09da6b9402d393d2b447d1f1f60f524ed1d99e0b5780300227979afb437d09R39